src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.add(user);
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        return Optional.ofNullable(userDao.findByEmail(email))
+                .orElseThrow(() -> new RuntimeException("Can't find user with this email"));
```

Comment:
Optional.ofNullable should not be used in this method, because the findByEmail method already returns an optional.